### ISSUE TITLE
Add support for Python dependency groups

### ISSUE THREAD

**Issue by redeboer** [prefix-dev/pixi#2256]
### Problem description

[PEP 735](https://peps.python.org/pep-0735/) recently [got accepted](https://discuss.python.org/t/pep-735-dependency-groups-in-pyproject-toml/39233/312) (see https://github.com/prefix-dev/pixi/issues/2211#issuecomment-2406948981). It can be used as a mechanism for defining Python dependencies for the developer environment without exposing them in the built distribution. I expect this will be quickly adopted by Python package managers, see e.g. https://github.com/python-poetry/poetry/issues/9751, https://github.com/astral-sh/uv/issues/8090, https://github.com/pypa/pip/issues/12963.

Would it be possible to include dependency groups in the Pixi lock file and recognize it in features, similarly like [how Pixi handles `optional-dependencies`](https://pixi.sh/v0.32.1/advanced/pyproject_toml/#optional-dependencies)?

**Comment by tdejager** [comment]
Yeah I think we should support, we could make an environment per dependency-group, or something else.

The most interesting thing is this part of the PEP is:

> The mutual compatibility of Dependency Groups is not guaranteed. For example, the Data Science example above shows conflicting versions of scikit-learn. Therefore, installing multiple locked dependency groups in tandem may require that tools apply additional constraints or generate additional lockfile data. These problems are considered out of scope for this PEP.
> As two examples of how combinations might be locked:
>   * A tool might require that lockfile data be explicitly generated for any combination to be considered valid
>   * Poetry implements the requirement that all Dependency Groups be mutually compatible, and generates only one locked version. (Meaning it finds a single solution, rather than a set or matrix of solutions.)

If you want to mix and match dependency groups in the same environment that would pose some problems.

**Comment by sirosen** [comment]
Hi there, PEP 735 author here! 👋

> If you want to mix and match dependency groups in the same environment that would pose some problems.

To make sure that this is clear, the spec doesn't say anything to forbid a _tool_ from having tighter constraints on dependency groups. For example, I'm aware that poetry requires that all dependency groups are mutually compatible.

So you don't need to support this if it poses a problem -- but that does put you in a position of having to tell your users what the behavior is (e.g., "it will fail when you try to lock dependencies").

It's possible that some users will misread the spec as saying that "tools are required to support this usage". But it's much more open than that, "tools should think about the fact that the spec doesn't protect against groups conflicting."

I hope that perspective helps!
(Also, I only learned about pixi a few months ago, but it looks very cool. Awesome work!)

**Comment by olivier-lacroix** [comment]
@tdejager , I think the dependency groups being composable, they should rather be mapped to features, as the current extras are.

**Comment by olivier-lacroix** [comment]
Logged a similar request against the [pyproject-toml crate](https://github.com/PyO3/pyproject-toml-rs/issues/23). Better to handle deserialisation there.

**Comment by tdejager** [comment]
@sirosen thank you for the reply and kind words :)! Great to see you here!

@olivier-lacroix I see that makes sense! We would still need to think if we want to instantiate some of these features by default in an evironment. Not sure yet!
